### ISSUE TITLE
Write integration tests for versioning features

### ISSUE THREAD

**Issue by NateWr** [pkp/pkp-lib#4888]
A list of UI steps for which tests should be written to prevent regressions in the versioning/publication features.

- [x] can edit data in publication tab
- [x] author can not edit data in publication tab
- [x] can schedule and publish version
- [x] version appears in frontend
- [x] can not edit published version
- [x] can create new version
- [x] can switch versions
- [x] can edit unpublished version
- [x] change author/galley/metadata
- [x] can publish second version
- [x] second version appears in frontend
- [x] frontend can navigate between versions
- [x] when viewing old version, an "outdated version" notice appears
- [x] "outdated version" notice includes link that navigates to latest article landing page
- [ ] when viewing old version, Google Scholar metatag `pdf_url` always goes to latest version of PDF galley
- [x] check author/galley/metadata
- [x] when viewing old version of PDF/HTML galley, an "outdated version" notice appears
- [x] "outdated version" notice in PDF/HTML galley includes link that navigates to latest article landing page
- [x] frontend shows correct current version
- [x] can unpublish version
- [x] frontend shows correct current version
- [x] frontend doesn't show unpublished version
- [ ] can edit previously published version
- [ ] can republish version
- [ ] frontend shows correct current version
- [x] frontend shows all published versions
- [x] change url path between versions and url works
- [x] change url path for galleys between versions and urls for galleys and files work
- [ ] change section, category, issue between versions and check listings in category/issue match current version
- [ ] a section editor with a "recommend only" stage assignment can not publish, unpublish or version the publication (API)
- [x] a section editor with a "recommend only" stage assignment does not see the publish, unpublish and create version buttons

**Comment by NateWr** [comment]
Well, this was an odyssey! But the fact that I made it through at all shows how much more I'm able to accomplish with these testing tools. :+1:

These PRs implement most of the suggested tests. I ran out of steam a bit but can dive back in to get the rest at a later date. I spent quite a bit of time trying to work out some nicer ways to run multiple tests against a submission -- mostly without a lot of luck.

One of the things I wanted to do was to find ways to work with test data that might live elsewhere (so we could use it for UI Library). @asmecher have a look at [this pattern](https://github.com/pkp/ojs/pull/2666/files#diff-5850d99222f6fc0533c9842582d2a2aaR12-R17) to load submission data. It was the only way I could figure out how to load a submission object, then be able to [update it with the ID](https://github.com/pkp/pkp-lib/pull/5586/files#diff-9593faa52e480476d0f5eef99f197fbbR172-R177) so that we can use the ID in subsequent tests.

Cypress suggests this is a hack and we should be using fixtures instead, but I've been every which way and back trying to use fixtures and aliases to accomplish this, and I'm just not sure it's possible (see my [stackoverflow question[(https://stackoverflow.com/questions/60510906/use-and-update-data-across-more-than-one-test-in-cypress)). Cypress really wants fixtures to remain static throughout the tests. But having the submission ID can help us shave off some time with redundant `findSubmissionAsEditor()` calls.

Let me know if you think that's alright or want to pursue something else. I'd be happy to replicate it across other submissions as well.

I also renamed John Mwandenga to Alan Mwandenga so that it runs first. I'm hoping that putting this one first will reduce the time-to-failure for tests, since it runs through every step of the workflow into publishing and versioning. Over time it might make sense to move more edge cases into this submission so that test failures are caught more quickly.

PRs:
https://github.com/pkp/pkp-lib/pull/5586
https://github.com/pkp/ui-library/pull/79
https://github.com/pkp/ojs/pull/2666

**Comment by asmecher** [comment]
Yes, interesting problem... This may not be a recommended approach but it's concise and reads clearly to me. I don't love doubling up the file count (`.spec.js` and `.json`) but reusability in the UI library is sufficient justification. My only request is that you keep the filenames consistent (`amwandenga` vs. `mwandenga`). I don't think ~~John~~Alan will mind the name change.

**Comment by NateWr** [comment]
The PRs involved a couple of minor bug fixes so I went ahead and merged for release in 3.2.0-1. I ended up removing the `mwandenga.json` data file. We can roll that method out later if/when we decide to refactor the tests or make common sample data with the UI Library.

I'm satisfied with the test coverage for now so I will close this, even though there are a couple of recommended tests that are still not implemented.
